src/hist/plot.py

```diff
@@ -72,26 +73,29 @@ def _expr_to_lambda(expr: str) -> Callable:
         varnames.append(tokval)
     varnames = list(OrderedDict.fromkeys([name for name in varnames if name != "x"]))
     lambdastr = f"lambda x,{','.join(varnames)}: {expr}"
-    return eval(lambdastr)
+    return eval(lambdastr)  # type: ignore
 
 
 def _curve_fit_wrapper(
-    func: Callable,
-    xdata: ArrayLike,
-    ydata: ArrayLike,
-    yerr: ArrayLike,
+    func: Callable[..., Any],
+    xdata: np.ndarray,
```

Comment:
ArrayLike can be a simple number, so `xdata[stuff]` was not valid. Either use np.asarray, or just require arrays.